services/federator/src/Federator/ExternalServer.hs

```diff
@@ -85,21 +117,43 @@ callLocal' req@Request {..} = do
   case HTTP.statusCode resStatus of
     200 -> pure $ maybe mempty LBS.toStrict resBody
     404 ->
-      case WaiError.label <$> (decode =<< resBody) of
+      case Wai.label <$> (decode =<< resBody) of
         Just "no-endpoint" -> throwInward IInvalidEndpoint (cs $ "component " <> show component <> "does not have an endpoint " <> cs validatedPath)
         _ -> throwInward IOther (cs $ show resStatus)
     code -> do
       let description = "Invalid HTTP status from component: " <> Text.pack (show code) <> " " <> Text.decodeUtf8 (HTTP.statusMessage resStatus) <> " Response body: " <> maybe mempty cs resBody
       throwInward IOther description
 
-routeToInternal :: (Members '[Service, Embed IO, Polysemy.Error ServerError, TinyLog, Polysemy.Reader RunSettings] r) => SingleServerT info Inward (Sem r) _
-routeToInternal = singleService (Mu.method @"call" callLocal)
+routeToInternal ::
+  (Members '[Service, Embed IO, Polysemy.Error ServerError, TinyLog, DiscoverFederator, Polysemy.Reader RunSettings] r) =>
+  Maybe ByteString ->
+  SingleServerT info Inward (Sem r) _
```

Comment:
At one point you had the return type spelled out, but then put back `_`. Are you planning to put back the type?

Reply:
I think `_` is fine, that's what the library docs suggest you do.